core/src/BibleEngine.class.ts

```diff
@@ -575,6 +588,48 @@ export class BibleEngine {
         return { ...book, ...updates };
     }
 
+    async downloadBooks(books: IBibleBookEntity[], fileUrl: string, version: BibleVersionEntity) {
+        const responses = await this.getBooksForDownload(books, fileUrl, version.uid);
+        const entityManager = await this.pDB;
+        for (const response of responses) {
+            const content: any = response;
+            await this.addBookWithContent(version, content, { entityManager });
+        }
+    }
+
+    async versionIsDownloaded(versionUid: string, fileHostUrl: string) {
+        const version = await this.getVersion(versionUid);
+        if (!version) return false;
+        if (version.dataLocation === 'remote') return false;
+        const booksInDb = await this.getBooksForVersionUid(versionUid);
+        const booksNotDownloaded = booksInDb.filter(
+            book => book.dataLocation === 'importing' || book.dataLocation === 'remote'
+        );
+        if (booksNotDownloaded.length) return false;
+        const totalBooks = await this.getBookIndexFile(versionUid, fileHostUrl);
+        return booksInDb.length === totalBooks.length;
+    }
+
+    async getBooksForDownload(books: any[], fileHostUrl: string, versionUid: string) {
+        const responses = await Promise.all(
+            books.map((book: any) => {
+                const bookContentsUrl = `${fileHostUrl}/${versionUid}/${book.osisId}.json`;
```

Comment:
i would recommend fetching this from the API as a first step and only introduce caching when really necessary. Last time i checked, there was no endpoint for fetching a whole book but adding that is only a one-liner.

Reply:
agreed. removed this function